pkg/operator/operator.go

```diff
@@ -811,6 +811,31 @@ func (op *Reporting) newPrometheusConn(promConfig promapi.Config) (prom.API, err
 	return prom.NewAPI(client), nil
 }
 
+func (op *Reporting) handleExpiredReport(report *metering.Report, now time.Time) error {
+	// check if the Report is being deleted already
+	if report.DeletionTimestamp != nil {
+		return fmt.Errorf("report was already marked for deletion")
+	}
+	// check if the Report is past its retention time
+	if reportExpired := isReportExpired(op.logger, report, now); !reportExpired {
+		op.logger.Debugf("the %s Report in the %s namespace has not reached the expiration date", report.Name, report.Namespace)
+		return nil
+	}
+	// check if the Report is used by any other Report or ReportQuery, if not delete it
+	if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+		op.deleteReport(report)
+		newDeletionTimestamp := metav1.Now()
+		report.SetDeletionTimestamp(&newDeletionTimestamp)
+		op.meteringClient.MeteringV1().Reports(report.Namespace).
+			Delete(context.TODO(), report.Name, metav1.DeleteOptions{})
```

Comment:
We should definintely catch an error is the delete call fails.

```suggestion
		err := op.meteringClient.MeteringV1().Reports(report.Namespace).Delete(context.TODO(), report.Name, metav1.DeleteOptions{})
		if err != nil {
		    return err
		}
```

Reply:
✅